src/field_map_change.c

```diff
@@ -241,57 +244,53 @@ void sub_020531A0 (FieldSystem * fieldSystem)
     gCoreSys.unk_65 = fieldSystem->unk_74->unk_00_12;
 }
 
-void sub_020531C0 (FieldSystem * fieldSystem, BOOL param1)
+void FieldMapChange_UpdateGameData (FieldSystem * fieldSystem, BOOL noWarp)
 {
-    int v0 = fieldSystem->unk_1C->unk_00;
-    UnkStruct_0203A790 * v1 = sub_0203A790(fieldSystem->saveData);
+    int mapId = fieldSystem->location->mapId;
+    FieldOverworldState * fieldState = SaveData_GetFieldOverworldState(fieldSystem->saveData);
 
-    sub_02055414(fieldSystem);
-    sub_0203F1C4(fieldSystem);
+    Sound_ClearSpecialBGM(fieldSystem);
+    FieldSystem_ClearLocalFlags(fieldSystem);
 
-    if (!param1) {
-        sub_02070494(fieldSystem);
+    if (!noWarp) { 
+        FieldSystem_InitFlagsWarp(fieldSystem);
     } else {
-        sub_02070430(fieldSystem);
+        FieldSystem_InitFlagsNoWarp(fieldSystem);
     }
 
-    sub_0206B0D4(SaveData_Events(fieldSystem->saveData));
+    FieldEvents_ResetVSSeeker(SaveData_GetVarsFlags(fieldSystem->saveData));
 
-    if (!param1) {
+    if (!noWarp) {
         sub_020559DC(fieldSystem);
     }
 
-    if (!param1) {
+    if (!noWarp) {
         sub_02027F50(sub_02027860(fieldSystem->saveData));
     }
```

Comment:
Do these branches really need to be separate?

Reply:
Checksum fails when trying to combine